src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -367,6 +373,14 @@ private bool TryRegister(SafeSocketHandle socket, out Interop.Error error)
             return error == Interop.Error.SUCCESS;
         }
 
+        private bool TryUnregister(SafeSocketHandle socket, out Interop.Error error)
+        {
+            error = Interop.Sys.TryChangeSocketEventRegistration(_port, socket,
+                Interop.Sys.SocketEvents.Read | Interop.Sys.SocketEvents.Write,
+                Interop.Sys.SocketEvents.None, socket.DangerousGetHandle());
+            return error == Interop.Error.SUCCESS;
```

Comment:
This happens automagically in the kernel when the socket is closed. And it doesn't solve the race issue between the epoll thread and the thread that creates/closes sockets. See https://github.com/dotnet/runtime/pull/36115#issuecomment-626569425.